tests/index.test.ts

```diff
@@ -494,21 +494,26 @@ describe('PTY', { repeats: 500 }, () => {
   });
 });
 
-describe('cgroup opts', () => {
+describe.only('cgroup opts', () => {
+  const CG_ROOT = '/sys/fs/cgroup';
+  const CG_TEST_SLICE = `${CG_ROOT}/test.slice`;
+
   beforeEach(async () => {
     if (!IS_DARWIN) {
-      // create a new cgroup with the right permissions
-      await exec("sudo cgcreate -g 'cpu:/test.slice'");
-      await exec(
-        'sudo chown -R $(id -u):$(id -g) /sys/fs/cgroup/cpu/test.slice',
-      );
+      const CG_ROOT = '/sys/fs/cgroup';
+      const SLICE = 'test.slice';
+      const SLICE_DIR = path.join(CG_ROOT, SLICE);
+
+      await exec(`sudo mkdir -p ${SLICE_DIR}`);
+      await exec(`sudo chown -R $(id -u):$(id -g) ${SLICE_DIR}`);
+      await exec(`echo ${process.pid} | sudo tee ${SLICE_DIR}/cgroup.procs`);
     }
   });
 
   afterEach(async () => {
     if (!IS_DARWIN) {
       // remove the cgroup
-      await exec('sudo cgdelete cpu:/test.slice');
+      await exec(`sudo cgdelete cpu:/test.slice`);
```

Comment:
caveat: you need to store the original cgroup in which `${process.pid}` lived in, and remove it from that list of cgroup processes for you to be able to delete the cgroup itself (you remove it from the cgroup by virtue of putting that process back into the cgroup it originally belonged to.

also, does this work

```suggestion
      const CG_ROOT = '/sys/fs/cgroup';
      const SLICE = 'test.slice';
      const SLICE_DIR = path.join(CG_ROOT, SLICE);
      await exec(`sudo rmdir ${SLICE_DIR}`);
```

?

the other consideration is that you do need to ensure that every node process executes exactly one test (no thread paralellism). if the thread is running multiple processes, we'll need to make `SLICE` be the test name or something unique.

Reply:
Sounds good! Yes `rmdir` works if we move `${process.pid}` back to its original cgroup.